src/cast_core/src/CrudActor.cpp

```diff
@@ -1117,19 +1117,24 @@ struct CommitTransactionOperation : public BaseOperation {
 
 struct WithTransactionOperation : public BaseOperation {
     WithTransactionOperation(const Node& opNode,
-                             bool onSession,
+                             bool,
```

Comment:
Is it possible to take in this parameter as a pointer to detect the difference between unset and false? I'm assuming that's why you retrieved it from the opNode below.

Reply:
@rtimmons I've just pushed [some commits](https://github.com/mongodb/genny/pull/998/files/553b05f80aa3bad3a6faa16ee2fca4062d70067b..643f008fcf06a9e7380a55a994ea0f2fd8c9cdd0) to remove all unused parameter from transaction-related operations. However, this requires quite a lot of plumbing changes. 

@thessem We sure can take in this parameter as a pointer instead of a boolean, but 1) it would require a lot of unrelated changes because of generic programming using templates (specifically, all `bool _onSession` in this file must be changed to `const Node& _onSession` including the templates, and their usages must be changed accordingly) and 2) I'm not sure if there's much difference between having the constructor taking a pointer to the node as a parameter, and the constructor getting the node by itself.

I did try to remove the usage of templates in this file, but it doesn't seem to be possible because C++ doesn't allow having typenames (and hence constructors) as values in an unordered_map (unlike Python for example). So the alternative would be massive if-else blocks everywhere the actual constructors of operations are invoked.